### PR TITLE
refactor: chunk -> share

### PR DESCRIPTION
**Pull request by rootulp**
Rename instances of `chunk` to `share`.

## Motivation

The rsmt2d repo uses chunks and shares interchangeably and we increase readability if one term was used consistently. A few people have expressed favor for share over chunk because of it's broader usage in celestia-node and celestia-app.

See https://github.com/celestiaorg/rsmt2d/issues/215

Note: There aren't occurrences of shard or buffer that need renaming in this repo. Shard is used frequently but it seems applicable to https://github.com/filecoin-project/dagstore/blob/master/docs/design.md#shards and not meant to refer to a share.

### CODE REVIEW

**Review by ramin — APPROVED** [review]

**Review by walldiss — APPROVED** [review]

**Comment by rootulp** [comment]
This PR needs a docs codeowner approval because it touches docs/adr. cc: @adlerjohn @liamsi 

Screenshot 2024-03-06 at 9 28 18 AM

Ref: https://github.com/celestiaorg/celestia-node/blob/32bc7a973b2f0b730c8ec1c2f7d6be5e5484e85b/.github/CODEOWNERS#L12

**Review by liamsi — APPROVED** [review]

**Pull request merged**
